js.go

```diff
@@ -267,16 +267,16 @@ const (
 	TraceReceived TraceOperation = 1
 )
 
-// TraceCB is called to trace API interactions for the JetStream Context
-type TraceCB func(op TraceOperation, subj string, payload []byte, hdr Header)
+// ClientTrace can be used to trace API interactions for the JetStream Context.
+type ClientTrace struct {
+	RequestSent      func(subj string, payload []byte, hdr Header)
+	ResponseReceived func(subj string, payload []byte, hdr Header)
```

Comment:
We could expand this to make them more granular for example, to only trace consumer lookups or for when adding retries make it possible to log an event for when there is a backoff, etc...